### PR TITLE
revouninstaller: Add version 2.2.0

### PR DESCRIPTION
**Pull request by schance995**
- Closes #562

See <https://github.com/lukesampson/scoop-extras/pull/562#issuecomment-636460531>.
Removed UAC registry evasion as it's not required for the app to function. Autoupdate reuses download link so the version # is properly updated.

### CODE REVIEW

File: bucket/revouninstaller.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
+    "homepage": "https://www.revouninstaller.com/products/revo-uninstaller-free/",
+    "license": "Freeware",
+    "description": "Uninstall software and remove unwanted programs easily",
+    "version": "2.2.0",
```
Comment (Ash258) [inline bucket/revouninstaller.json:5]:
```suggestion
    "version": "2.2.0",
    "description": "Software uninstaller with support for additional cleanup of registries/files",
    "homepage": "https://www.revouninstaller.com/products/revo-uninstaller-free/",
    "license": "Freeware",
```

File: bucket/revouninstaller.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    "shortcuts": [
+        [
+            "RevoUPort.exe",
+            "RevoUninstaller"
+        ]
+    ],
+    "checkver": {
+        "url": "https://www.revouninstaller.com/products/revo-uninstaller-free",
+        "regex": "Version: ([\\d.]+)"
+    },
```
Comment (Ash258) [inline bucket/revouninstaller.json:24]:
```suggestion
    "checkver": "Version: ([\\d.]+)",
```

File: bucket/revouninstaller.json [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+            "RevoUPort.exe",
+            "revouninstaller"
+        ]
+    ],
+    "shortcuts": [
+        [
+            "RevoUPort.exe",
+            "RevoUninstaller"
+        ]
+    ],
```
Comment (Ash258) [inline bucket/revouninstaller.json:20]:
```suggestion
    "shortcuts": [
        [
            "RevoUPort.exe",
            "Revo Uninstaller"
        ]
    ],
```

File: bucket/revouninstaller.json [hunk 4]
```diff
@@ -0,0 +1,28 @@
+{
+    "homepage": "https://www.revouninstaller.com/products/revo-uninstaller-free/",
+    "license": "Freeware",
+    "description": "Uninstall software and remove unwanted programs easily",
+    "version": "2.2.0",
+    "hash": "f543d9e0627f4170674310f5759498bc65a08446c3c2693f9771493c9d3063f5",
+    "url": "https://www.revouninstaller.com/download-free-portable.php#/dl.zip",
```
Comment (Ash258) [inline bucket/revouninstaller.json:7]:
```suggestion
    "url": "https://www.revouninstaller.com/download-free-portable.php#/dl.zip",
    "hash": "f543d9e0627f4170674310f5759498bc65a08446c3c2693f9771493c9d3063f5",
```

**Review by Ash258 — CHANGES_REQUESTED** [review]

**Review by Ash258 — CHANGES_REQUESTED** [review]
- Persist dat files and settings
- Disable autoudpate

image

File: bucket/revouninstaller.json [hunk 5]
```diff
@@ -6,14 +6,40 @@
...
+            "        Set-Content \"$dir\\$file\" $content -Encoding Ascii",
+            "    }",
+            "}",
+            "CreateFile \"ctrlbars.dat\"",
+            "CreateFile \"un_report.dat\"",
+            "CreateFile \"winapp_report.dat\"",
+            "CreateFile \"settings.ini\""
+        ]
+    },
+    "post_install": "(Get-Content \"$dir\\settings.ini\") -replace 'AU on startup=1','AU on startup=0' | Set-Content \"$dir\\settings.ini\"",
```
Comment (Ash258) [inline bucket/revouninstaller.json:23]: Why post?

**Comment by schance995** [comment]
Config files are written after first run of app, I think best thing to do is notify user. Feel free to edit the notes if they are too long.

File: bucket/revouninstaller.json (hunk 5 above)
Comment (Ash258) [inline bucket/revouninstaller.json:23]: - pre_install everything
- Do this: `'file1', 'file2' | ForEach-Object { if (!....) { new-item } }`

File: bucket/revouninstaller.json [hunk 6]
```diff
@@ -6,14 +6,40 @@
...
         [
             "RevoUPort.exe",
             "Revo Uninstaller"
         ]
     ],
+    "persist": [
+        "ctrlbars.dat",
+        "un_report.dat",
+        "winapp_report.dat",
+        "settings.ini"
```
Comment (Ash258) [inline bucket/revouninstaller.json:34]: sort

File: bucket/revouninstaller.json [hunk 7]
```diff
@@ -6,14 +6,40 @@
...
     "checkver": "Version: ([\\d.]+)",
     "autoupdate": {
         "url": "https://www.revouninstaller.com/download-free-portable.php#/dl.zip"
-    }
+    },
+    "notes": [
+        "This app requires admin privileges to run.",
+        "The first time you run the app, close it right away so the app can create config files.",
+        "Then it is highly recommended to disable autoupdate in the app (Options -> Disable autoupdate), or in settings.ini"
+    ]
```
Comment (Ash258) [inline bucket/revouninstaller.json:44]: Notes are essential information to user. It needs to be at top of manifest, after license

File: bucket/revouninstaller.json [hunk 8]
```diff
@@ -6,14 +6,40 @@
...
+    ],
     "checkver": "Version: ([\\d.]+)",
     "autoupdate": {
         "url": "https://www.revouninstaller.com/download-free-portable.php#/dl.zip"
-    }
+    },
+    "notes": [
+        "This app requires admin privileges to run.",
+        "The first time you run the app, close it right away so the app can create config files.",
+        "Then it is highly recommended to disable autoupdate in the app (Options -> Disable autoupdate), or in settings.ini"
```
Comment (Ash258) [inline bucket/revouninstaller.json:43]: Completely nonse to user as it will be disable automatically.

File: bucket/revouninstaller.json [hunk 9]
```diff
@@ -6,14 +6,40 @@
...
+        "settings.ini"
+    ],
     "checkver": "Version: ([\\d.]+)",
     "autoupdate": {
         "url": "https://www.revouninstaller.com/download-free-portable.php#/dl.zip"
-    }
+    },
+    "notes": [
+        "This app requires admin privileges to run.",
+        "The first time you run the app, close it right away so the app can create config files.",
```
Comment (Ash258) [inline bucket/revouninstaller.json:42]: What??

**Review by Ash258 — CHANGES_REQUESTED** [review]

**Comment by schance995** [comment]
Persist files are created on first run of app. I tried editing autoupdate on install but it won't work without writing the whole `settings.ini`. I think the best thing to do is to notify user about autoupdate.

**Comment by schance995** [comment]
I see I misinterpreted autoupdate, apologies.

File: bucket/revouninstaller.json [hunk 10]
```diff
@@ -6,21 +6,15 @@
...
-            "CreateFile \"un_report.dat\"",
-            "CreateFile \"winapp_report.dat\"",
-            "CreateFile \"settings.ini\""
-        ]
-    },
-    "post_install": "(Get-Content \"$dir\\settings.ini\") -replace 'AU on startup=1','AU on startup=0' | Set-Content \"$dir\\settings.ini\"",
+    "pre_install": [
+        "'ctrlbars.dat', 'un_report.dat', 'settings.ini', 'winapp_report.dat' | ForEach-Object {",
+        "    if (!(Test-Path \"$persist_dir\\$_\")) {",
+        "        Write-Host 'File' $_ 'does not exists. Creating.' -f Yellow",
```
Comment (Ash258) [inline bucket/revouninstaller.json:12]:
```suggestion
```

File: bucket/revouninstaller.json [hunk 11]
```diff
@@ -6,21 +6,15 @@
...
+        "        Set-Content \"$dir\\$_\" $content -Encoding Ascii",
```
Comment (Ash258) [inline bucket/revouninstaller.json:13]: There is no $content.

Use New-Item

File: bucket/revouninstaller.json [hunk 12]
```diff
@@ -6,21 +6,15 @@
...
+        "    }",
+        "}",
+        "(Get-Content \"$dir\\settings.ini\") -replace 'AU on startup=1','AU on startup=0' | Set-Content \"$dir\\settings.ini\""
```
Comment (Ash258) [inline bucket/revouninstaller.json:16]: DRY.

- Just save the AU into regex capture group and then reference it in replace
- Always use encoding ascii in set-content

**Review by Ash258 — CHANGES_REQUESTED** [review]

**Comment by pratikpc** [comment]
Closing as schance995:revouninstaller has been deleted.

**Pull request closed without merging**